equistore/io.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+
+from ._c_lib import _get_library
+from .block import TensorBlock
+from .data import _is_numpy_array, _is_torch_array
+from .labels import Labels
+from .tensor import TensorMap
+
+
+def load(path: str, use_numpy=False) -> TensorMap:
+    """Load a previously saved :py:class:`equistore.TensorMap` from the given path.
+
+    :py:class:`equistore.TensorMap` are serialized using numpy's ``.npz``
+    format, i.e. a ZIP file without compression (storage method is ``STORED``),
+    where each file is stored as a ``.npy`` array. See the C API documentation
+    for more information on the format.
+
+    :param path: path of the file to load
+    :param use_numpy: should we use numpy or the native implementation? The main
+        difference is that numpy should be able to save more dtypes for the data
+        than the native implementation, which is limited to float64.
+    """
+    if use_numpy:
+        return _read_npz(path)
+    else:
+        lib = _get_library()
+        return TensorMap._from_ptr(lib.eqs_tensormap_load(path.encode("utf8")))
+
+
+def save(path: str, tensor: TensorMap, use_numpy=False):
+    """Save the given :py:class:`equistore.TensorMap` to a file at ``path``.
+
+    :py:class:`equistore.TensorMap` are serialized using numpy's ``.npz``
+    format, i.e. a ZIP file without compression (storage method is ``STORED``),
+    where each file is stored as a ``.npy`` array. See the C API documentation
+    for more information on the format.
+
+    :param path: path of the file where to save the data
+    :param tensor: tensor to save
+    :param use_numpy: should we use numpy or the native implementation? The main
+        difference is that numpy should be able to save more dtypes for the data
+        than the native implementation, which is limited to float64.
+    """
+    if not path.endswith(".npz"):
+        # warning?
```

Comment:
I would add the ending explicitly then.

Reply:
I don't really like "magic" behavior like this, I would rather send a warning and have the user fix its code.

Reply:
A warning and saving the file without the npz ending?

Reply:
Hmm, I checked how numpy handles this, and it automagically adds the extension to the file ... Let's go for a warning and saving the file with an extension to follow numpy then.